imagepipeline/src/main/java/com/facebook/imagepipeline/producers/DecodeProducer.java

```diff
@@ -439,6 +443,10 @@ protected synchronized boolean updateDecodeJob(EncodedImage encodedImage, boolea
       return ret;
     }
 
+    private boolean isNotPartialData(EncodedImage encodedImage) {
+      return encodedImage.getEncodedCacheKey() != null;
```

Comment:
an intermediate result either came from network fetcher under a poor network connection, or disk cache when media-variants could be found with the media-id set on a request, as fresco only cache a image onto disk cache if it received the full raw bytes of the image( or isLast == true, checkout ```DiskCacheWriteConsumer```) and any encoded image came from MediaVariationsFallbackProducer with a media-variant satisfies the request then it should has a valid cache key, you cant checkout DiskCacheReadProducer